### DISCUSSION TITLE
Dynamic routing + getServerSideProps weird behavior on Vercel

### DISCUSSION THREAD

**Discussion by Djiit** [vercel/next.js#15369] [Help]
Hey 👋 

I'm using the recent `getServerSideProps` on a 9.4 NextJS app hosted on Vercel.

When I develop locally, I can reach a page, let say `/pages/test/[test].ts` without any problem. `test` is a random id, let's say "1". I made a simple page that display the `test` value. Works great.

When I push it to Vercel, this page (`/test/1`) is not found (404). It should be rendered by a function. I don't even see a function invokation in the Vercel dashboard... What is weird is I can reach e.g. https://nextbot-git-test-ssg.lourd.vercel.app/app/test/[test] (the link should work, it's a test page without any auth). So, meh, is that normal ?

Did I miss something obvious ?

Thanks for you help

edit: more infos : 

- This `/app`is just a custom prefix handled by a custom function until basePath lands (9.4.5?)
- I could use query params. Pretty sure it will work; but not the intended UX.

**Comment by jamesmosier** [comment]
If I had to guess, this is likely where your problem is:

> This /app is just a custom prefix handled by a custom function until basePath lands

but unfortunately that'd just be a guess. 

Assuming that is coming via a custom server (correct me if I'm wrong) then Vercel does not work with a custom server. https://github.com/vercel/next.js/issues/9397#issuecomment-556215227

**Comment by Djiit** [comment]
Hey James,

Thanks for your answer.
No, I'm not using any custom server, thanks for pointing this out. Here is the function I'm refering to : 
```ts
export const getPrefixedPath = (path) =>
  process.env.ASSET_PREFIX !== "" ? `${process.env.ASSET_PREFIX}${path}` : path;
```

Every other pages work great, FWIW.

**Reply by jamesmosier** [reply]
Where does that function run?

**Reply by Djiit** [reply]
In pages, when I need to get static assets or links.

**Comment by Djiit** [comment]
Here is a reproduction example : https://github.com/Djiit/nextjs-repro-ssr / https://nextjs-repro-ssr-git-master.djiit.vercel.app/test/100

Hu, ok, this is working without my function. Let me try to add it back.

**Reply by Djiit** [reply]
Yes, it works with the function (it's just string manipulation : https://nextjs-repro-ssr.vercel.app/test/1)

**Comment by jamesmosier** [comment]
`ASSET_PREFIX` is not the same a basePath unfortunately. So prefixing your paths with that will not work as you'd expect. 

Asset prefixes are for serving assets from a CDN, for example. https://nextjs.org/docs/api-reference/next.config.js/cdn-support-with-asset-prefix

As you mentioned, you'll want to keep an eye out for the experimental basePath feature coming soon.

Aside from that, there is no way to do basePaths today with Next (v9.4.4) like you are looking for.

**Reply by Djiit** [reply]
Sorry to insist, but I guess the variable ASSET_PREFIX is adding confusion. This could work with anything else. say, MY_PREFIX.

See it working here : https://nextjs-repro-ssr.vercel.app/app/

**Reply by Djiit** [reply]
(here the prefix is /app)

**Comment by Djiit** [comment]
So after testing, it appears that adding a `vercel.json` with a `routes` configuration breaks the pages that uses getServerSideProps. Full repro is available here : 
- repo : https://github.com/Djiit/nextjs-repro-ssr
- failing page: https://nextjs-repro-ssr.vercel.app/app/test/1
- page not supposed to be here : https://nextjs-repro-ssr.vercel.app/app/test/[id]
